### PR TITLE
Macros prefixation

### LINKED ISSUE #153: Prefix project macros to prevent conflicts
Preprocessing macros defined by the project should be prefixed by `CORENEURON_` to prevent possible conflicts.

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Fix #153.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -107,7 +107,9 @@ include(Findnmodl)
 # ispc support
 if (ENABLE_ISPC_TARGET)
     enable_language(ISPC)
-    add_definitions("-DISPC_INTEROP=1")
+    if (CORENEURON_ISPC_INTEROP)
```
Comment (tristan0x) [inline CMakeLists.txt]: I don't think this extra `if` statement is correct.

Comment (tapaswenipathak) [reply CMakeLists.txt]: Just interested in knowing, the variable(precisely flag/dir) should be inside an `if`, no?

Comment (tristan0x) [reply CMakeLists.txt]: `CORENEURON_ISPC_INTEROP` CMake variable does not exist and is never set.
According to the code in [master branch](https://github.com/BlueBrain/CoreNeuron/blob/master/CMakeLists.txt#L108-L112), the `CORENEURON_ISPC_INTEROP` macro should be set if `ENABLE_ISPC_TARGET` CMake variable is True.

Comment (tapaswenipathak) [reply CMakeLists.txt]: :cool:

Comment (tapaswenipathak) [reply CMakeLists.txt]: :cool: ok, I just wanted that as an option as every CMake variable should be present as an option and under an `if`.

**Comment by tapaswenipathak** [comment]
@tristan0x ping, do you think I should add anything else in the PR? Please let me know.

**Comment by tapaswenipathak** [comment]
@tristan0xdo you think I can enhance anything else in the PR? Please let me know.

**Comment by tapaswenipathak** [comment]
@tristan0x do you think I can enhance anything else in the PR?

**Comment by tapaswenipathak** [comment]
👋

**Comment by pramodk** [comment]
I am closing this because the macro renaming changes needs to be done at two places outside this project : https://github.com/BlueBrain/mod2c and https://github.com/BlueBrain/nmodl. Given the complete code-reorganisation of this repo in #229, it would be better to re-do this.

**Pull request closed without merging**